pre_commit_hooks/leading_tabs_remover.py

```diff
@@ -0,0 +1,48 @@
+import argparse
+import re
+
+MAX_REPLACE = 10
+TAB_WIDTH = 4
+
+
+def fix_file(filename, tab_width=TAB_WIDTH, max_replace=MAX_REPLACE):
+    with open(filename, 'r') as fd:
+        original_lines = fd.readlines()
+    new_lines = []
+    changed = False
+    regex = re.compile(r'^\s*\t')
+    for line in original_lines:
+        replace_count = 0
+        while regex.match(line) and replace_count < max_replace:
+            changed = True
+            replace_count += 1
+            line = line.replace('\t', ' ' * tab_width, 1)
+        if not replace_count < max_replace:
+            line = line.strip('\r\n')
+            raise RuntimeError('Reached max tab replacements for one line '
```

Comment:
Not sure this is really a concern worth the added complexity here.

Alternative ideas if you're not too tired with this module already:

```python
m = re.match('^(\s+)', line)
if m:
    leading_spaces = m.groups()[0]
    line = leading_spaces.replace('\t', '    ') + line.lstrip()
```

Or it could be done without regex:
```python
leading_spaces = line[:-len(line.lstrip())]
line = leading_spaces.replace('\t', '    ') + line.lstrip()
```

Reply:
I'll use one of these suggestions. The max replacements comes from an earlier issue where my first iteration of the fixer had an infinite loop due to my simplistic implementation here.